onnx_xla/onnx_xla_interface.cc

```diff
@@ -267,3 +277,79 @@ ONNXIFI_SYMBOL_NAME(onnxReleaseGraph)(onnxGraph graph) {
   }
   ONNXIFI_CATCH_EXCPETION();
 }
+
+//Initialize event by creating EventControl object on the heap
+ONNXIFI_PUBLIC ONNXIFI_CHECK_RESULT onnxStatus ONNXIFI_ABI ONNXIFI_SYMBOL_NAME(
+    onnxInitEvent)(onnxBackend backend, onnxEvent* event) {
+  try {
+    if (!backend)  {
+      return ONNXIFI_STATUS_INVALID_BACKEND;
+    }
+    if (!event)  {
+      return ONNXIFI_STATUS_INVALID_POINTER;
+    }
+
+    *event = (onnxEvent) new EventControl();
+    return ONNXIFI_STATUS_SUCCESS;
+  }
+  ONNXIFI_CATCH_EXCPETION();
+
+
+}
+
+//Signal Event by changing the signalled boolean under mutex hold
+ONNXIFI_PUBLIC ONNXIFI_CHECK_RESULT onnxStatus ONNXIFI_ABI ONNXIFI_SYMBOL_NAME(
+    onnxSignalEvent)(onnxEvent event)  {
+  try {
+    auto *e = reinterpret_cast<EventControl *>(event);
+    if (!event)  {
+      return ONNXIFI_STATUS_INVALID_EVENT;
+    }
+    {
+       std::lock_guard<std::mutex> lk(e->m);
+       if (e->signalled)  {
+         return ONNXIFI_STATUS_INVALID_STATE;
+       }
+       e->signalled = true;
+    }
+    e->cv.notify_all();
+    return ONNXIFI_STATUS_SUCCESS;
+  }
+  ONNXIFI_CATCH_EXCPETION();
```

Comment:
The macro `ONNXIFI_CATCH_EXCPETION ` is not self contained, which makes it looks very weird. Could you change it to use a function (that takes a lambda as input) for doing this try catch return error code stuff?

Reply:
Passing lambda to self-contained try-catch macro now; previous structure was copied for the tensor-rt PR, however